### PR TITLE
Upgrade to JDK 1.8

### PR DESCRIPTION
**Pull request by lewismc**
This issue addresses #125

### CODE REVIEW

**Comment by lewismc** [comment]
For some reason the TravisCI config is not being picked up.

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 language: java
 
+script:
+  - jdk_switcher use oraclejdk8
```
Comment (jnioche) [inline .travis.yml]: Have you tried 

```
jdk:
  - oraclejdk8
```

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+  - mvn install javadoc:aggregate
```
Comment (jnioche) [inline .travis.yml]: what's this one for?

File: src/main/java/crawlercommons/domains/EffectiveTldFinder.java [hunk 3]
```diff
@@ -74,13 +74,13 @@ public static EffectiveTldFinder getInstance() {
         return instance;
     }
 
-    public boolean initialize(InputStream effective_tld_data_stream) {
```
Comment (jnioche) [inline src/main/java/crawlercommons/domains/EffectiveTldFinder.java]: Are _ not allowed in Java8?

**Comment by jnioche** [comment]
Did you run `mvn java-formatter:format` before committing?

File: .travis.yml (hunk 2 above)
Comment (lewismc) [inline .travis.yml]: This defines the targets to run... we always wish to build the Javadoc target as this lets us know if the Javadoc is JDK8 compliant.

File: src/main/java/crawlercommons/domains/EffectiveTldFinder.java (hunk 3 above)
Comment (lewismc) [inline src/main/java/crawlercommons/domains/EffectiveTldFinder.java]: Yes... but they are flagged as warn's for the Javac. It is not Java convention.

**Comment by lewismc** [comment]
> Did you run mvn java-formatter:format before committing?

Actually no I did not.. I will do that right now.
Thanks for review @jnioche

**Comment by lewismc** [comment]
@jnioche can you please review again at your earlier opportunity. Thank you.

File: .travis.yml (hunk 2 above)
Comment (jnioche) [inline .travis.yml]: ok, thanks. I assumed the javadoc was checked by Travis

**Comment by jnioche** [comment]
Merged, thanks @lewismc

**Pull request merged**